server/datastore/mysql/software.go

```diff
@@ -152,8 +152,8 @@ func getOrGenerateSoftwareIdDB(ctx context.Context, tx sqlx.ExtContext, s fleet.
 	var existingId []int64
 	if err := sqlx.SelectContext(ctx, tx,
 		&existingId,
-		`SELECT id FROM software WHERE name = ? AND version = ? AND source = ? AND bundle_identifier = ?`,
-		s.Name, s.Version, s.Source, s.BundleIdentifier,
+		`SELECT id FROM software WHERE name = ? AND version = ? AND source = ?`,
```

Comment:
Remove bundle identifier here so that the select is covered by the index we've got. Also, to get the id, bundle identifier doesn't really matter.

Below, I don't think we'll be in the UPDATE portion ever. If we do, would LastInsertedId error out?

Reply:
Hm, actually, doing this makes a test fail... so I'm reverting, and checking the explain, it seems to be using the index anyway... so 🤐